x/evm/keeper/abci.go

```diff
@@ -44,5 +52,33 @@ func (k *Keeper) EndBlock(ctx context.Context) error {
 	bloom := ethtypes.BytesToBloom(k.GetBlockBloomTransient(infCtx).Bytes())
 	k.EmitBlockBloomEvent(infCtx, bloom)
 
+	feeCollectorAddr := authtypes.NewModuleAddress(authtypes.FeeCollectorName)
+	balance := k.bankKeeper.GetBalance(sdkCtx, feeCollectorAddr, k.GetParams(sdkCtx).EvmDenom)
+	if balance.IsZero() {
+		return nil
+	}
+
+	chainFeeSplitterAddress := common.HexToAddress(k.GetParams(sdkCtx).ChainFeeSplitterAddress)
```

Comment:
Just to double-check: If `k.GetParams(sdkCtx).ChainFeeSplitterAddress` is either an empty string `""` or a 20-byte zero address, `common.HexToAddress` will produce `common.Address{}` and we will skip the transfer. Is that correct?

Reply:
Yes, it's very important to not transfer fees anywhere from the `fee_collector` account if a `chainFeeSplitterAddress` was not set / is zero.